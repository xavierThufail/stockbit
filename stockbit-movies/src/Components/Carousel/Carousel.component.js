import React from 'react';
import { Carousel as AntdCarousel } from 'antd';

import imageUrl from '../../Constants/carousel';
import styles from './Carousel.style';

const Carousel = ({ height = '200px', margin, width }) => {
  return (
    <AntdCarousel autoplay>
      {imageUrl.map((url, index) => (
        <div key={index}>
          <div style={{...styles(url), height, width, margin}} />
        </div>
      ))}
    </AntdCarousel>
  );
};

export default Carousel;
