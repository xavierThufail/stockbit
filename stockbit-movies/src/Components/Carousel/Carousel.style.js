const styles = (url) => ({
  backgroundImage: url,
  backgroundPosition: 'center',
  backgroundSize: 'cover',
  borderRadius: '15px'
});

export default styles;
