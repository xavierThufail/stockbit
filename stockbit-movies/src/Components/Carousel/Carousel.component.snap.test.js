import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Carousel from './Carousel.component';

describe('Carousel component snap test', () => {
  it('Render Carousel, should match snapshot', () => {

    const { container } = render(<Carousel />);

    expect(container).toMatchSnapshot();
  });
});
