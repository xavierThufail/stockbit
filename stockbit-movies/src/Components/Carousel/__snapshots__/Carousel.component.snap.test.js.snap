// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component snap test Render Carousel, should match snapshot 1`] = `
<div>
  <div
    class="ant-carousel"
  >
    <div
      class="slick-slider slick-initialized"
      dir="ltr"
    >
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-1"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  style="background-image: url(https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/10af0b39-a57e-4f10-aeb1-fd388299ed5b/dd30sej-ba00f8a2-fdcf-48ad-8681-e507010cba43.png?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7InBhdGgiOiJcL2ZcLzEwYWYwYjM5LWE1N2UtNGYxMC1hZWIxLWZkMzg4Mjk5ZWQ1YlwvZGQzMHNlai1iYTAwZjhhMi1mZGNmLTQ4YWQtODY4MS1lNTA3MDEwY2JhNDMucG5nIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmZpbGUuZG93bmxvYWQiXX0.pWIARDMHQJbaWljid6OPz2QhIqDTW5DPZW3cXG-iUEA); background-position: center; background-size: cover; border-radius: 15px; height: 200px;"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  style="background-image: url(https://wallpaperboat.com/wp-content/uploads/2019/06/cool-deadpool-01.jpg); background-position: center; background-size: cover; border-radius: 15px; height: 200px;"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  style="background-image: url(https://i.pinimg.com/originals/49/f1/b1/49f1b154f5e606051fd2818e061dc8c0.jpg); background-position: center; background-size: cover; border-radius: 15px; height: 200px;"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  style="background-image: url(https://www.xtrafondos.com/wallpapers/superman-con-traje-negro-5945.jpg); background-position: center; background-size: cover; border-radius: 15px; height: 200px;"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="3"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  style="background-image: url(https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/10af0b39-a57e-4f10-aeb1-fd388299ed5b/dd30sej-ba00f8a2-fdcf-48ad-8681-e507010cba43.png?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7InBhdGgiOiJcL2ZcLzEwYWYwYjM5LWE1N2UtNGYxMC1hZWIxLWZkMzg4Mjk5ZWQ1YlwvZGQzMHNlai1iYTAwZjhhMi1mZGNmLTQ4YWQtODY4MS1lNTA3MDEwY2JhNDMucG5nIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmZpbGUuZG93bmxvYWQiXX0.pWIARDMHQJbaWljid6OPz2QhIqDTW5DPZW3cXG-iUEA); background-position: center; background-size: cover; border-radius: 15px; height: 200px;"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="4"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  style="background-image: url(https://wallpaperboat.com/wp-content/uploads/2019/06/cool-deadpool-01.jpg); background-position: center; background-size: cover; border-radius: 15px; height: 200px;"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="5"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  style="background-image: url(https://i.pinimg.com/originals/49/f1/b1/49f1b154f5e606051fd2818e061dc8c0.jpg); background-position: center; background-size: cover; border-radius: 15px; height: 200px;"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="6"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  style="background-image: url(https://www.xtrafondos.com/wallpapers/superman-con-traje-negro-5945.jpg); background-position: center; background-size: cover; border-radius: 15px; height: 200px;"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="7"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  style="background-image: url(https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/10af0b39-a57e-4f10-aeb1-fd388299ed5b/dd30sej-ba00f8a2-fdcf-48ad-8681-e507010cba43.png?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7InBhdGgiOiJcL2ZcLzEwYWYwYjM5LWE1N2UtNGYxMC1hZWIxLWZkMzg4Mjk5ZWQ1YlwvZGQzMHNlai1iYTAwZjhhMi1mZGNmLTQ4YWQtODY4MS1lNTA3MDEwY2JhNDMucG5nIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmZpbGUuZG93bmxvYWQiXX0.pWIARDMHQJbaWljid6OPz2QhIqDTW5DPZW3cXG-iUEA); background-position: center; background-size: cover; border-radius: 15px; height: 200px;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <ul
        class="slick-dots slick-dots-bottom"
        style="display: block;"
      >
        <li
          class="slick-active"
        >
          <button>
            1
          </button>
        </li>
        <li
          class=""
        >
          <button>
            2
          </button>
        </li>
        <li
          class=""
        >
          <button>
            3
          </button>
        </li>
        <li
          class=""
        >
          <button>
            4
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
